crates/sui-graphql-rpc/src/server/watermark_task.rs

```diff
@@ -4,27 +4,32 @@
 use crate::data::{Db, DbConnection, QueryExecutor};
 use crate::error::Error;
 use crate::metrics::Metrics;
+use crate::types::chain_identifier::ChainIdentifier;
 use async_graphql::ServerError;
 use diesel::{ExpressionMethods, OptionalExtension, QueryDsl};
 use std::mem;
 use std::sync::Arc;
 use std::time::Duration;
 use sui_indexer::schema::checkpoints;
 use tokio::sync::{watch, RwLock};
+use tokio::time::Interval;
 use tokio_util::sync::CancellationToken;
 use tracing::{error, info};
 
 /// Watermark task that periodically updates the current checkpoint and epoch values.
 pub(crate) struct WatermarkTask {
     /// Thread-safe watermark that avoids writer starvation
     watermark: WatermarkLock,
+    chain_identifier: ChainIdentifierLock,
     db: Db,
     metrics: Metrics,
     sleep: Duration,
     cancel: CancellationToken,
     sender: watch::Sender<u64>,
     receiver: watch::Receiver<u64>,
 }
+#[derive(Clone, Default)]
```

Comment:
nit: whitespace above this line.

```suggestion

#[derive(Clone, Default)]
```